packages/react-native-worklets/src/shareables.ts

```diff
@@ -139,6 +139,14 @@ function makeShareableCloneRecursiveNative<T>(
     return cloneBigInt(value) as ShareableRef<T>;
   }
 
+  if (value === undefined) {
+    return cloneUndefined() as ShareableRef<T>;
```

Comment:
I believe we don't need to call `cloneUndefined()` each time – instead, we can just store only one instance of it in some const and always return the same instance. In fact, we don't even need a function, but it's okay for now.